firstvoices/backend/views/stats_views.py

```diff
@@ -0,0 +1,157 @@
+from datetime import timedelta
+
+from django.utils import timezone
+from drf_spectacular.utils import OpenApiResponse, extend_schema, extend_schema_view
+from rest_framework import viewsets
+from rest_framework.response import Response
+
+from backend.models import DictionaryEntry, Song, Story
+from backend.models.constants import Visibility
+from backend.models.dictionary import TypeOfDictionaryEntry
+from backend.models.media import Audio, Image, Video
+from backend.views import doc_strings
+from backend.views.api_doc_variables import site_slug_parameter
+from backend.views.base_views import FVPermissionViewSetMixin, SiteContentViewSetMixin
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="A list of statistics about a given site.",
+        responses={
+            200: Response,
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[site_slug_parameter],
+    ),
+)
+class StatsViewSet(SiteContentViewSetMixin, FVPermissionViewSetMixin, viewsets.ViewSet):
+    """API endpoint that returns statistics about the specified site."""
+
+    def list(self, request, *args, **kwargs):
```

Comment:
I would like some advice on how to best optimize this endpoint if possible, as there are currently 200+ SQL requests showing on a site stats page with no models